x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/TransportRollupAction.java

```diff
@@ -144,8 +149,24 @@ protected void masterOperation(
         ActionListener<AcknowledgedResponse> listener
     ) {
         String sourceIndexName = request.getSourceIndex();
-        IndexMetadata sourceIndexMetadata = state.getMetadata().index(sourceIndexName);
+
+        final IndicesAccessControl indicesAccessControl = threadContext.getTransient(AuthorizationServiceField.INDICES_PERMISSIONS_KEY);
+        if (indicesAccessControl != null) {
+            final IndicesAccessControl.IndexAccessControl indexPermissions = indicesAccessControl.getIndexPermissions(sourceIndexName);
+            if (indexPermissions != null) {
+                boolean hasDocumentLevelPermissions = indexPermissions.getDocumentPermissions().hasDocumentLevelPermissions();
+                boolean hasFieldLevelSecurity = indexPermissions.getFieldPermissions().hasFieldLevelSecurity();
+                if (hasDocumentLevelPermissions || hasFieldLevelSecurity) {
+                    listener.onFailure(
+                        new ElasticsearchException(
+                            "Rollup forbidden for index [" + sourceIndexName + "] with document level or field level security settings."
```

Comment:
I used "Rollup" here instead of "Downsample" just to be coherent with other error messages.

Reply:
Aren't all messages in `TransportRollupAction` class say  "Downsample" instead of "Rollup? If not I think we should change them (in this or a follow up PR)